### ISSUE TITLE
typescript errors

### ISSUE THREAD

**Issue by gbishop** [josephmbustamante/phaser3-typescript-starter-kit#2]
I get numerous errors from typescript when I attempt to run the project. I'll list the first below and perhaps you can tell me what is going wrong.

```
ERROR in /home/gb/Sync/phaser3-typescript-starter-kit/src/main.ts
./src/main.ts
[tsl] ERROR in /home/gb/Sync/phaser3-typescript-starter-kit/src/main.ts(25,37)
      TS2345: Argument of type 'GameConfig' is not assignable to parameter of type 'Phaser.Types.Core.GameConfig'.
  Types of property 'input' are incompatible.
    Type 'boolean | InputConfig' is not assignable to type 'boolean | Phaser.Types.Core.InputConfig'.
      Type 'InputConfig' is not assignable to type 'boolean | InputConfig'.
        Type 'InputConfig' is not assignable to type 'Phaser.Types.Core.InputConfig'.
          Types of property 'keyboard' are incompatible.
            Type 'boolean | KeyboardInputConfig' is not assignable to type 'boolean | Phaser.Types.Core.KeyboardInputConfig'.
              Type 'KeyboardInputConfig' is not assignable to type 'boolean | KeyboardInputConfig'.
                Type 'KeyboardInputConfig' is not assignable to type 'Phaser.Types.Core.KeyboardInputConfig'.
                  Types of property 'capture' are incompatible.
                    Type 'number' is not assignable to type 'number[]'.
```

I can run another typescript phaser project I found so I think my environment is OK. 

Any help appreciated.

**Comment by limeforadime** [comment]
I'm getting the same issue. I tried replacing "GameConfig" with "Phaser.Types.Core.GameConfig", but I'm getting other compilation problems after that.

Also seeing error "TS2339: Property 'resize' does not exist on type 'Game'"

**Comment by josephmbustamante** [comment]
There was a fairly large refactoring to how types were incorporated in Phaser 3.17 once they were added to the actual project repository and not something you had to grab separately. I think I just never took the time to fix some of the issues that came with that upgrade. I'll spend some time looking into it and fully updating the starter kit to be compatible with the newest version of Phaser.

**Comment by tailot** [comment]
@josephmbustamante I tried to solve this problem in the pull request https://github.com/josephmbustamante/phaser3-typescript-starter-kit/pull/4

Regards
Vincenzo

**Comment by josephmbustamante** [comment]
Hey everyone, this should all be fixed now. I had a branch with all the changes that has been sitting around for awhile, but just realized it never actually got pushed up, my apologies. I updated to the latest releases of Phaser (3.20) and TypeScript (3.7). I also added the player movement discussed in the blog post to the repository, and fleshed out the README a bit more. Thanks for reporting these issues, and a big thank you to @tailot for taking the time to submit a PR. Closing this.
